### ISSUE TITLE
Creating content types dynamically

### ISSUE THREAD

**Issue by mihirlibran** [perfectsense/brightspot-cms#1202]
Is there any way I can create content types dynamically in Brightspot?

I went to Admin -> Settings -> Dashboard and in the field for Common Content Settings, I clicked on Add Item -> New Object Type. I was able to create a new Content Type from here. However, I'm trying to achieve the same through Java code. Is there any way I can create one through Java code apart from creating a new class which extends Content?

Specifically, I was trying the following approach:

`
ObjectType objectType = new ObjectType();
`
`
objectType.setDisplayName("Sample Content Type");
`
`
//This doesn't work because there is no empty constructor for ObjectField!
`
`
ObjectField objectField = new ObjectField();
` 

Is this approach correct or can it be achieved in some other way?

Thank you!

**Comment by rhino88** [comment]
Hi mihirlibran,

That is the correct way to accomplish what you are suggesting.

After saving the ObjectType, you will need to execute the following snippet:

```
DatabaseEnvironment env = Database.Static.getDefault().getEnvironment();
env.refreshGlobals();
env.refreshTypes();
env.refreshGlobals();
env.refreshTypes();
```

**Comment by mihirlibran** [comment]
Hi Rhino88,

Thank you for your response. 

I was able to create a content type dynamically. But it does not show up in the 'Filters' dropdown (the one that shows after clicking on the search box). I've set com.psddev.cms.db.Content as one of the superclasses, and I've also set cms.content.searchable as an item in the groups.
Is it supposed to behave this way or do I need to do something more to have it show up in the Filters dropdown?

Thanks!
